### PR TITLE
Don't wrap negative indices twice (fix #2826)

### PR DESCRIPTION
**Pull request by nicowilliams**
Resolves #2826

### CODE REVIEW

File: tests/jq.test [hunk 1]
```diff
@@ -2004,3 +2004,9 @@ walk(1)
 walk(select(IN({}, []) | not))
 {"a":1,"b":[]}
 {"a":1}
+
+# Issue #2826
+.[-5]
```
Comment (itchyny) [inline tests/jq.test:2009]: We can merge the test at L251 with the following query; `[.[range(-4; 4)]]`.

**Comment by nicowilliams** [comment]
But now:

```
./jq -cn '[0, 1, 2]|path(.[-5])'
[-2]
```

**Comment by itchyny** [comment]
Can we just drop support of index wrapping for the coming release and consider how pick should work on arrays later?

File: src/execute.c [hunk 2]
```diff
@@ -695,18 +695,23 @@ jv jq_next(jq_state *jq) {
...
       }
       jv v = jv_get(t, jv_copy(k));
       if (jv_is_valid(v)) {
-        path_append(jq, k, jv_copy(v));
+        path_append(jq, norm_k, jv_copy(v));
         stack_push(jq, v);
+        jv_free(k);
       } else {
+        jv_free(norm_k);
         jv_free(k);
```
Comment (emanuele6) [inline src/execute.c]: This is calling `jv_free(k)` in both branches; also you can just use `k` for the `jv_get()` at line 708.

**Comment by nicowilliams** [comment]
> Can we just drop support of index wrapping for the coming release and consider how pick should work on arrays later?

The current version of this PR does not break `pick/1`.

**Pull request closed without merging**